force-app/main/default/classes/LeadTriggerHandler.cls

```diff
@@ -0,0 +1,33 @@
+public class LeadTriggerHandler{
+    public static void createLeadTask(List<Lead> leadRecords, Map<Id, Lead> oldLeadRecords){
+        List<Task> insertTasks = new List<Task>();
+        if(leadRecords.size() > 0){
```

Comment:
Not strictly necessary to check for list size > 0. :) For loops can handle empty lists.